### PR TITLE
feat(ui): Icon Search in Storybook

### PR DESCRIPTION
**Pull request by vuluongj20**
**1. New page for icon search, with:**
- Fuzzy, keyword-based search
    Screen Shot 2021-12-02 at 4 30 20 PM
- Popover panels with prop selectors and copiable code snippets
    Screen Shot 2021-12-02 at 4 31 10 PM

**2. Missing icons** 
Adding icons that are in the design system on Figma but not yet in `static/app/icons`, e.g. `IconText`.

**3. File Icon story**
Moving it into the Components section, as FileIcon seems like more of a component than an asset (SVG, PNG).

### CODE REVIEW

File: static/app/icons/iconUnlock.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+type Props = React.ComponentProps<typeof SvgIcon>;
+
+const IconUnlock = React.forwardRef(function IconUnlock(
+  props: Props,
+  ref: React.Ref<SVGSVGElement>
+) {
+  return (
+    <SvgIcon {...props} ref={ref}>
+      <g clipPath="url(#a)">
```
Comment (Jesse-Box) [inline static/app/icons/iconUnlock.tsx:13]: @vuluongj20 I double-checked this icon and the paths are bigger than the dimensions of the SVG itself, probably why you have this `clipPath` prop.

Check it out.

Screen Shot 2021-12-03 at 16 10 38

File: static/app/icons/iconSpaceship.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+type Props = React.ComponentProps<typeof SvgIcon>;
+
+const IconSpaceship = React.forwardRef(function IconSpaceship(
+  props: Props,
+  ref: React.Ref<SVGSVGElement>
+) {
+  return (
+    <SvgIcon {...props} ref={ref}>
+      <defs>
+        <clipPath id="a">
```
Comment (Jesse-Box) [inline static/app/icons/iconSpaceship.tsx:14]: @vuluongj20 Same goes here

File: static/app/icons/iconSend.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+type Props = React.ComponentProps<typeof SvgIcon>;
+
+const IconSend = React.forwardRef(function IconSend(
+  props: Props,
+  ref: React.Ref<SVGSVGElement>
+) {
+  return (
+    <SvgIcon {...props} ref={ref}>
+      <defs>
+        <clipPath id="a">
```
Comment (Jesse-Box) [inline static/app/icons/iconSend.tsx:14]: @vuluongj20 you get the point lol

**Review by Jesse-Box — DISMISSED** [review]
If you wanna add these new icons to the library we first need to tweak them on the design side first.

**Comment by vuluongj20** [comment]
> If you wanna add these new icons to the library we first need to tweak them on the design side first.

@Jesse-Box Good catches! Added a couple of commits to address the issues you raised: removing the unnecessary clipPaths (https://github.com/getsentry/sentry/pull/30392/commits/46420803bc748d5006b12895eca5b4741b89da3e), and resizing IconLock/IconUnlock (https://github.com/getsentry/sentry/pull/30392/commits/1baf41caface59893e473ac76b3d6b52015b5b62).

**Comment by evanpurkhiser** [comment]
Can we do this PR without adding new icons for now?

**Comment by vuluongj20** [comment]
> Can we do this PR without adding new icons for now?

Yeah makes sense - just removed them!

**Comment by billyvg** [comment]
The popover is pretty cool - I got a bit confused though as I expected the icon that I clicked to change when for example, toggled isCircled:

image

**Review by billyvg — APPROVED** [review]

**Pull request merged**